salt/_modules/caasp_docker.py

```diff
@@ -40,44 +40,53 @@ def get_registries_certs(lst, default_port=5000):
 
     LOG.debug('Finding certificates in: %s', lst)
     for registry in lst:
-        url = registry.get('url')
-        cert = registry.get('cert', '')
-        if len(cert) > 0:
-
-            # parse the name as an URL or "host:port", and return <HOST>[:<PORT>]
-            hostname, port = _get_hostname_and_port(url)
-            host_port = hostname
-            if port:
-                host_port += ":" + str(port)
-
-            LOG.debug('Adding certificate for: %s', host_port)
-            certs[host_port] = cert
-
-            if port:
-                if port == default_port:
-                    # When using the standar port (5000), if the user introduces
-                    # "my-registry:5000" as a trusted registry, he/she will be able
-                    # to do "docker pull my-registry:5000/some/image" but not
-                    # "docker pull my-registry/some/image".
-                    # So we must also create the "ca.crt" for "my-registry"
-                    # as he/she could just access "docker pull my-registry/some/image",
-                    # and Docker would fail to find "my-registry/ca.crt"
-                    name = hostname
+        try:
+            url = registry.get('url')
+
+            cert = registry.get('cert', '')
```

Comment:
So, if the `cert` is `None` it will prevail as this `''` fallback is just in case the `cert` entry does not exist, so the `''` case never gets used. Another way to get around the issue would be to do `registry.get('cert', '') or ''`. That way the cert would never be a `None` (note that the `''` on the `registry.get` could very well be optional, but in any case, it's good to have).

Wouldn't that fix the issue alone?

Reply:
The `if cert` in the next line will fix the issue...

Reply:
Isn't it better to fail an orchestration rather than make it work and have some stuff not configured properly (and hence causing harder to debug problems?)

Reply:
At the end, the difference will be 1) orchestration succedes but no certificate is added and an error is logged or 2) orchestration just fails. Developers usually prefer 1), while users prefer 2). Both situations will lead to users filling a new issue in Bugzilla, and they are both easy to debug.

In this case I prefer 1) too (as I think it is an overkill to  abort an orchestration by a failure on the certificates), but I guess this is more a philoshophical question...

Reply:
Fair enough, I don't have a strong opinion on that. I'm fine leaving things as they currently are implemented (catch exception)